### ISSUE TITLE
Porting audio renderer from mpc-be

### ISSUE THREAD

**Issue by adipose** [clsid2/mpc-hc#601]
The built-in audio renderer in mpc-be is much simpler and follows a more traditional design model for a filter.  Is it worth porting it over?

What features from sanear are important?

The main reason I ask is the code for sanear is a bit hard to follow, but it seems to have some advanced syncing logic.  I'm just not sure if it achieves anything that mpc audio renderer doesn't?

**Comment by clsid2** [comment]
I honestly don't know if Sanear has any benefits compared to the BE renderer.

If you want to port it, that is totally ok! We don't necessarily need to replace Sanear. We can include both and decide about removing Sanear in future. Simply rename the internal audio renderer to "Sane Audio Render (Sanear)". Preferably move the Sanear settings to its own property page similar to the BE renderer. Then add button to access settings of selected renderer.

It's code doesn't necessarily need to be ported/included. It could also be included as an external binary and loaded similar as LAV Filters are. Same could be done for MPC VR.

**Comment by ghost** [comment]
I use MpcAudioRenderer.ax from MPC-BE's external filters zip file so I can get multichannel LPCM audio working. With the default MPC-HC renderer the channels don't map to 5.1. I also have to use an older release of MpcAudioRenderer.ax (MPC-BE zip file from the 2017) so that the player doesn't hang when playing a media file sitting on an older external drive and waiting for it to spin up.

**Comment by demolitioner** [comment]
@sobake 
What you mean by "renderer the channels don't map to 5.1"? In my case MPC-HC perfectly fine decode and map 5.1 channels wav / ac3 / dts to 5.1 outputs.

**Comment by ghost** [comment]
@demolitioner 
@clsid2 

I'm not quite sure why but multichannel audio sources are downmixed to stereo. Only when I switch to the MPC Audio Renderer can I hear all the channels output from their respective speakers. Curiously, I specifically have to untick the "Use system layout channels" option in MPC Audio Renderer to get the multichannel audio, otherwise it downmixes to stereo again.

The other option that I like in MPC Audio Renderer is the "Release device when idle" option which lets me pause, rather than close the player in WASAPI exclusive mode and listen to something else.

I should mention that the speakers are connected to the analog motherboard outputs, not HDMI or any other digital output.

**Comment by demolitioner** [comment]
@sobake
Are your system sound configured as 5.1? Open win+r, command mmsys.cpl, set your sound device as "Set Default" and "Configure" it as 5.1 in this window. Other mediaplayers, like foobar2000, output 5.1 normal?

P.S. Select "Speakers", click "Configure" in this window and set "5.1 Surround".

**Comment by ghost** [comment]
@demolitioner 
This is what I see

![sound](https://user-images.githubusercontent.com/69406351/92952986-82495500-f426-11ea-8316-3d024198f15a.PNG)
![speakers](https://user-images.githubusercontent.com/69406351/92952907-604fd280-f426-11ea-9689-9b12999018a9.PNG)
![balance](https://user-images.githubusercontent.com/69406351/92952910-62b22c80-f426-11ea-83ac-57ee814ac25e.PNG)
